quantum/process_keycode/process_backlight.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright 2019
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "process_backlight.h"
+
+#include "backlight.h"
+
+bool process_backlight(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        case BL_ON:
+            if (record->event.pressed) {
```

Comment:
you could probably try hoisting the conditional outside of the switch. It would remove a lot of boilerplate.

```c
if (record->event.pressed) {
  switch(keycode) {
    case BL_ON:
      backlight_level(BACKLIGHT_LEVELS);
      return false;
    default:
      return true;
  }
  return true;
}
return true;
```

This should be equivalent and shorter.

Reply:
Down to 18636 bytes from 18690 🎉